main/src/test/scala/sbt/internal/ProjectNegSpec.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * sbt
+ * Copyright 2011 - 2017, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * Licensed under BSD-3-Clause license (see LICENSE)
+ */
+
+package sbt.internal
+
+import org.scalatest.FunSuite
+import sbt.internal.TestUtil._
+
+import scala.tools.reflect.ToolBoxError
+
+class ProjectNegSpec extends FunSuite {
+  def expectError(
+      errorSnippet: String,
+      compileOptions: String = "",
+      baseCompileOptions: String = s"-cp $toolboxClasspath",
+  )(code: String) = {
+    val errorMessage = intercept[ToolBoxError] {
+      eval(code, s"$compileOptions $baseCompileOptions")
+      println(s"Test failed -- compilation was successful! Expected:\n$errorSnippet")
+    }.getMessage
+    val userMessage =
+      s"""
+         |FOUND: $errorMessage
+         |EXPECTED: $errorSnippet
+      """.stripMargin
+    assert(errorMessage.contains(errorSnippet), userMessage)
+  }
+
+  test("Fail on self-referencing project") {
+    expectError("Project root cannot depend on itself") {
+      """
+        |import sbt._
+        |import Process._
+        |import Keys._
+        |import Project._
+        |
+        |val root = project in new java.io.File(".")
+        |dependsOn(root)
```

Comment:
`-> reload` did not work, so I am trying to write negative test using ScalaTest.

Most of sbt DSL seems not available on reflective compilation, unless DSL are properly imported as we do so in build settings on `.scala`. (I guess) 

`project` and `in` works fine,  if I add `import Project._` where those 2 DSL are defined.

However, I am not able to use `file` defined in `sbt` package object and `depensOn` defined in `sbt.dsl` package object
E.g., if `import sbt.dsl._` added,  test still fail with `cala.tools.reflect.ToolBoxError: reflective compilation has failed: not found: value dependsOn` error. 

@eed3si9n Am I missing something?

Reply:
sbt package object is in another subproject - https://github.com/sbt/sbt/blob/1.x/sbt/src/main/scala/package.scala